### PR TITLE
Backend improvements closes #95 and closes #96

### LINKED ISSUE #95: Increase code coverage on API's and inner structures

### LINKED ISSUE #96: Move some responsibilities away from DAO's

### PR DESCRIPTION
**Pull request by GijsWeterings**

### CODE REVIEW

File: src/main/java/me/moodcat/api/SongAPI.java [hunk 1]
```diff
@@ -157,19 +158,22 @@ private VAVector adjustSongVector(final ClassificationRequest classification, fi
...
     @AllArgsConstructor
+    @NoArgsConstructor
     protected static class ClassificationRequest {
 
         /**
          * The valence for the song.
          */
         @Getter
-        private final double valence;
+        @Setter
```
Comment (JWGmeligMeyling) [inline src/main/java/me/moodcat/api/SongAPI.java]: If you have both `@Setter` and `@Getter` on all attributes of the class, why not use `@Data` on the class?

File: src/main/java/me/moodcat/api/RoomAPI.java [hunk 2]
```diff
@@ -56,15 +58,11 @@ public RoomAPI(final RoomDAO roomDAO, final ChatDAO chatDAO) {
         VAVector targetVector = VAVector.createTargetVector(moods);
         List<Room> allRooms = roomDAO.listRooms(targetVector, limit);
 
-        if (limit > allRooms.size()) {
+        if (limit > allRooms.size() || limit < 1) {
```
Comment (JWGmeligMeyling) [inline src/main/java/me/moodcat/api/RoomAPI.java]: I would rather see a limit of `null` to be used. A limit of `0` is confusing, I would expect `0` results then, and not all.

**Comment by JeremybellEU** [comment]
The last commits have most likely broken some frontend API calls in RoomAPI, as they don't return Responses anymore but concrete classes. We have to investigate and ensure they still work before merging this branch.

Maybe it is also worth rebasing and squashing due to the sheer amount of commits regarding the average method.

The last 2 commits closes #95 and #96 .

**Comment by Azkle** [comment]
Just as @JeremybellEU said, the frontend has totally broke.
We can't send messages and there is no song playing in a room (it is in the overview,but not in the actual room), therefore I can't check SAMs and likeing.

Are you going to fix this or do we need to take a look on how we're gonna fix this as a team?

**Comment by JWGmeligMeyling** [comment]
(No, seriously :wink: )

File: src/main/java/me/moodcat/api/RoomAPI.java [hunk 3]
```diff
@@ -33,6 +39,11 @@
 public class RoomAPI {
 
     /**
+     * The number of miliseconds in a second, duh.
+     */
+    private static final int SECOND_OF_MILISECONDS = 1000;
```
Comment (GijsWeterings) [inline src/main/java/me/moodcat/api/RoomAPI.java]: milliseconds*

**Comment by Azkle** [comment]
It all works and the code seems correct.

**Comment by JeremybellEU** [comment]
Yes looks good, merging!

**Pull request merged**